app/lib/search_query_parser.rb

```diff
@@ -1,13 +1,13 @@
 # frozen_string_literal: true
 
 class SearchQueryParser < Parslet::Parser
-  rule(:term)      { match('[^\s":]').repeat(1).as(:term) }
+  rule(:term)      { match('[^\s":+-]').repeat(1).as(:term) }
   rule(:quote)     { str('"') }
   rule(:colon)     { str(':') }
   rule(:space)     { match('\s').repeat(1) }
   rule(:operator)  { (str('+') | str('-')).as(:operator) }
   rule(:prefix)    { term >> colon }
-  rule(:shortcode) { (colon >> term >> colon.maybe).as(:shortcode) }
+  rule(:shortcode) { (operator.maybe >> colon >> term >> colon.maybe).as(:shortcode) }
```

Comment:
Maybe an alternative would be to treat it as a `term` since the handling in the transformer is basically the same. This would avoid the need for the transformer code changes.